### ISSUE TITLE
Protocol matching does not reject classmethod in class where protocol uses an instance method

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6870]
```python
from typing import Protocol

class Proto(Protocol):
    def method1(self, a: int, b: int) -> float:
        ...

class Concrete:
    @classmethod
    def method1(self, a: int, b: int) -> float:
        return 0

x: Proto = Concrete() # Should generate an error
```

**Comment by erictraut** [comment]
I misunderstood the spec here. This isn't actually a problem. Closing.
